### PR TITLE
Fix order of sources, fix etag saving

### PR DESCRIPTION
**Pull request by FrozenAlex**

### CODE REVIEW

File: src/Data/DataGetter.cpp [hunk 1]
```diff
@@ -89,8 +90,8 @@ namespace SongDetailsCache {
         std::string line;
         while (std::getline(buf, line)) {
             if (line.starts_with("ETag: ")) {
-                auto tagStart = line.find_first_of('"') + 1;
-                auto tagLength = line.find_last_of('"') - tagStart;
+                auto tagStart = line.find_first_of('"');
```
Comment (RedBrumbler) [inline src/Data/DataGetter.cpp:92]: sorry what is this doing

File: src/Data/SongDetailsContainer.cpp [hunk 2]
```diff
@@ -60,20 +60,27 @@ namespace SongDetailsCache {
         }
 
         for (const auto& [src, _] : DataGetter::dataSources) {
-            auto db = DataGetter::UpdateAndReadDatabase(src).get();
-            if (!db.has_value()) {
-                break;
-            }
+            DEBUG("Trying source {}", src);
```
Comment (RedBrumbler) [inline src/Data/SongDetailsContainer.cpp]: please remove this one and just log if a source failed (at the catch below, just make it print the src there)

File: include/Data/DataGetter.hpp [hunk 3]
```diff
@@ -15,7 +16,7 @@ namespace SongDetailsCache {
                 std::shared_ptr<std::vector<uint8_t>> data;
             };
 
-            static const std::unordered_map<std::string, std::string> dataSources;
+            static const std::map<std::string, std::string> dataSources;
```
Comment (RedBrumbler) [inline include/Data/DataGetter.hpp]: Please keep this an unordered map and just reorder them to fix the order. put the one that should be checked first at the bottom, and last at the top. that's how unordered map works

File: src/Data/DataGetter.cpp (hunk 1 above)
Comment (RedBrumbler) [reply src/Data/DataGetter.cpp:92]: image
from what I can tell this should be parsing the tag already? unless the tag should include `"` at the begin and end?

Comment (FrozenAlex) [reply src/Data/DataGetter.cpp:92]: It should include it yeah

Comment (FrozenAlex) [reply src/Data/DataGetter.cpp:92]: Server does not understand it without the quotes and sends an entire file to us

**Pull request merged**